### PR TITLE
Removal of the "picture" field

### PR DESCRIPTION
**Pull request by AnnaLangner**
Removal of the "picture" field

### CODE REVIEW

File: script/main.py [hunk 1]
```diff
@@ -25,14 +25,17 @@ def init_argparser():
 
 def find_records(people_json):
   record_names_phone = []
-  record_names_with_dob =[]
+  record_names_picture = []
```
Comment (klangner) [inline script/main.py]: The logic for removing the records with pictures should be in the `remove_record_with_picture_from_json` function

**Review by klangner — CHANGES_REQUESTED** [review]

File: script/main.py (hunk 1 above)
Comment (AnnaLangner) [reply script/main.py]: Searching and removing a record should be in one function?

Comment (klangner) [reply script/main.py]: You are not searching here.
You are only removing records given some predicate (rule)
Search would be if you want to fins some records and at the same time you will also remove some of them.

Comment (AnnaLangner) [reply script/main.py]: Was it about something like this?

Comment (klangner) [reply script/main.py]: Yes that's better.

BTW I think it would be better to split commands in different functions. so main() will not be overloaded.
something like this:
```python
if args.operation == 'init':  
    init()
elif args.operation == 'percentage':
    percentage()
```

Comment (AnnaLangner) [reply script/main.py]: Do these functions move inside init ()?
``` 
    create_new_record_with_dob_in_json(dob_records)
    removes_special_characters_from_phone_and_cell_numbers(phone_records)
    remove_record_with_picture_from_json(people_json)
```

Comment (klangner) [reply script/main.py]: This question is part of the task :-)

Comment (AnnaLangner) [reply script/main.py]: it's done ;)

File: script/main.py [hunk 2]
```diff
@@ -238,13 +242,9 @@ def main():
   results = []
   people_json =  json.load(open("init\persons.json", encoding='utf-8'))
```
Comment (klangner) [inline script/main.py]: Do you need it here?

File: script/main.py [hunk 3]
```diff
@@ -129,22 +133,28 @@ def create_users_table(conn):
...
-def init_db(conn):  
+def init_db(conn, people_json):  
   '''Initializing the database'''    
 
   # create table
   if conn is not None:
     create_users_table(conn)
   else:
-    print("Error! cannot create the database connection.")
+    print("Error! cannot create the database connection.")
```
Comment (klangner) [inline script/main.py]: It is not enough to print error.
But I think in this case it is ok to assume that conn is not None. 
so this check would not be necessary

File: script/main.py [hunk 4]
```diff
@@ -129,22 +133,28 @@ def create_users_table(conn):
...
   '''Initializing the database'''    
 
   # create table
   if conn is not None:
     create_users_table(conn)
   else:
-    print("Error! cannot create the database connection.")
+    print("Error! cannot create the database connection.")  
+  
+  (phone_records, dob_records) = find_records(people_json)
```
Comment (klangner) [inline script/main.py]: Here before we will fix it.
Could write in plain English steps which are necessary to initialize database?
don't look at those functions yet, since they are not optimal.

**Review by klangner — CHANGES_REQUESTED** [review]

File: script/main.py (hunk 2 above)
Comment (AnnaLangner) [reply script/main.py]: nope :)

File: script/main.py (hunk 4 above)
Comment (AnnaLangner) [reply script/main.py]: 1. connect to json or api file
2. delete and change some records
3. create a connection with db
4. create a table
5. import of records to db

Was it about something like this?

Comment (klangner) [reply script/main.py]: Yes something like that. But those step are not correct. Eg. 
Do you really connect to json API?, 
Do you create connection here or get it as a parameter?
etc.

I think that this ability to write the correct step is something very important which you should learn.
When you have steps writing code is easy :-)
So lets try to improve it!

Comment (AnnaLangner) [reply script/main.py]: 1. take JSON file as parameter
2. add a new record to the JSON file
3. delete and change some records in the JSON file
4. create a connection with db
5. create a table
6. import of records to db

Comment (klangner) [reply script/main.py]: Nope :-)

* You don't need to do anything to take parameter
* You need to be more specific: What kind of record you want to add?
* do you really create connection here, do you create table

Write it is such way, that other person will understand what should be done.

Comment (AnnaLangner) [reply script/main.py]: 1. Use a JSON file as a parameter
2. Find the record in the JSON file
3. Add a new record to the JSON file that counts  down to birthday
4. Remove the photo from the records in the JSON file
5. Remove special characters from phone in JSON file
6. Create a table
7. Import of records from a JSON file

Comment (AnnaLangner) [reply script/main.py]: 1. Load a JSON file.
2. Convert the dictionary into a list of records and access individual fields.
3. Add an additional field to the records that calculate how many days are left until the next birthday.
4. Remove of the "photo" field from the records.
5. Remove of special characters from phone and cell fields in all records.
6. Create a table.
7. Insert data into the database.

Comment (klangner) [reply script/main.py]: Looks ok to me. The only questions left is:
What is: `access individual fields` in point 2?

As a next step try to rename functions and variables, so the function init() will looks like implementation of this points

Comment (AnnaLangner) [reply script/main.py]: is access to dob field, phone and cell

Comment (AnnaLangner) [reply script/main.py]: The changes have been made

File: script/main.py [hunk 5]
```diff
@@ -142,42 +142,42 @@ def insert_users_to_table(conn, users):
...
-    dob_age = dict_of_person['dob']['age']
-    dob_time_until_birthday = dict_of_person['dob']['time_until_birthday']
-    registered_date = dict_of_person['registered']['date']
-    registered_age = dict_of_person['registered']['age']
-    phone = dict_of_person['phone']
-    cell = dict_of_person['cell']
-    id_name = dict_of_person['id']['name']
-    id_value = dict_of_person['id']['value']
-    nat = dict_of_person['nat']
+  for dict_of_people in list_of_people:
```
Comment (klangner) [inline script/main.py]: Usually you don't add data structure name to the variable name:
So instead of `list_of_peoples` it is better to name it just `peoples`
`dict_of_people` => `person`

File: script/main.py [hunk 6]
```diff
@@ -142,42 +142,42 @@ def insert_users_to_table(conn, users):
...
+    dob_age = dict_of_people['dob']['age']
+    dob_time_until_birthday = dict_of_people['dob']['time_until_birthday']
+    registered_date = dict_of_people['registered']['date']
+    registered_age = dict_of_people['registered']['age']
+    phone = dict_of_people['phone']
+    cell = dict_of_people['cell']
+    id_name = dict_of_people['id']['name']
+    id_value = dict_of_people['id']['value']
+    nat = dict_of_people['nat']
     columns = [
```
Comment (klangner) [inline script/main.py]: btw instead of creating all this variables, you could just write it as:
```
columns = [
  dict_of_people['gender'],
 ...
]
```

Comment (klangner) [inline script/main.py]: Columns is wrong name for this list. You don't store columns here.
Better name: `record` old `fields`

File: script/main.py [hunk 7]
```diff
@@ -213,9 +213,9 @@ def insert_users_to_db(conn, people):
...
     ]
 
-    people.append(columns)      
+    users.append(columns)      
     
-  insert_users_to_table(conn, people)
+  insert_users_to_table(conn, users)
 
 
 def init_db(conn):
```
Comment (klangner) [inline script/main.py:182]: Does it work?

Comment (klangner) [reply script/main.py:182]: Please provide short function names

**Review by klangner — CHANGES_REQUESTED** [review]

File: script/main.py (hunk 7 above)
Comment (AnnaLangner) [reply script/main.py:182]: yes it work `python script/main.py init`

File: script/main.py [hunk 8]
```diff
@@ -23,25 +23,25 @@ def init_argparser():
...
-  record_names_phone = []
-  record_names_with_dob =[]
-  dict_json_to_list = people_json['results']
-  for dict_single_record in dict_json_to_list:
-    if 'phone' and 'cell' in dict_single_record:
-      record_names_phone.append(dict_single_record)
-    if 'dob' in dict_single_record:
-      record_names_with_dob.append(dict_single_record)
-  return (record_names_phone, record_names_with_dob)
+def convert_dict_to_list_access_phone_cell_dob_field(people):
```
Comment (klangner) [inline script/main.py]: Please make it shorter. Maybe: `extract_phone_numbers` or similar.
And what is `dob`? Maybe it is better to move it to separate function?

File: script/main.py [hunk 9]
```diff
@@ -129,122 +133,83 @@ def create_users_table(conn):
...
+  insert_users_to_table(conn, users)
+
+
 def init_db(conn):  
-  '''Initializing the database'''    
+  people =  json.load(open("init\persons.json", encoding='utf-8'))
+  (phone_fields, dob_fields) = convert_dict_to_list_access_phone_cell_dob_field(people)
+  add_field_time_until_birthday(dob_fields)
+  remove_special_characters_from_phone_numbers(phone_fields)
+  remove_field_with_picture_from_records(people)
```
Comment (klangner) [inline script/main.py]: What's the point of calling this function here?
No matter if you remove the pictures or not. The database will look the same.

**Review by klangner — CHANGES_REQUESTED** [review]

File: script/main.py (hunk 8 above)
Comment (AnnaLangner) [reply script/main.py]: "dob" is the field in the record the same as "telephone" or "cell"

File: script/main.py (hunk 9 above)
Comment (AnnaLangner) [reply script/main.py]: I have to call this function somewhere. If main () should be clean, the next place it should be is the init () function.
When I'm wrong, correct me.

Comment (klangner) [reply script/main.py]: No. The question was: Is this function doing anything useful?
It doesn't matter if you remove pictures or not. You will not add them to the database anyway

Comment (AnnaLangner) [reply script/main.py]: In fact, I didn't think about its meaning. The command was to delete it, I deleted it.

**Review by klangner — APPROVED** [review]
